### PR TITLE
Update the build for AS canary 4.1

### PR DESCRIPTION
**Pull request by stevemessick**
We had a lot of catching up to do.

@DaveShuckerow @devoncarew

### CODE REVIEW

**Comment by devoncarew** [comment]
Looking now (looks like compilation errors for 2020.1?: https://travis-ci.org/github/flutter/flutter-intellij/jobs/669923128?utm_medium=notification&utm_source=github_status)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
+# 45.0
+- Add support for 'flutter pub outdated' (#4444)
+- Improve how we calculate when to summarize flutter errors (#4447)
+- Fix an issue with discovering hyperlinks in test consoles (#4443)
```
Comment (devoncarew) [inline CHANGELOG.md:4]: These first three items (and the new supported platforms) would be good fodder for the blog.

File: src/io/flutter/utils/AndroidLocationProvider.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.utils;
+
+import com.android.tools.idea.gradle.dsl.model.BuildModelContext;
```
Comment (devoncarew) [inline src/io/flutter/utils/AndroidLocationProvider.java:8]: We may have issues w/ local development, for people developing against something like IntelliJ CE?

File: tool/plugin/lib/plugin.dart [hunk 3]
```diff
@@ -638,121 +638,161 @@ class BuildCommand extends ProductCommand {
...
+          source = processedFile.readAsStringSync();
+          files[processedFile] = source;
+          source = source.replaceAll(
+            'findGradleTarget', // 3 matches
+            'findImportTarget',
+          );
+          processedFile.writeAsStringSync(source);
+        }
+      } catch (ex) {
+        // Restore sources.
```
Comment (devoncarew) [inline tool/plugin/lib/plugin.dart:780]: Do we want this restore code to happen on an exception, or in a `finally`?

Comment (stevemessick) [reply tool/plugin/lib/plugin.dart:780]: It happens in a finally after the compiler is finished.

File: src/io/flutter/utils/AndroidLocationProvider.java (hunk 2 above)
Comment (stevemessick) [reply src/io/flutter/utils/AndroidLocationProvider.java:8]: Yes, Until we have time to devise a proper solution people can replace AndroidLocationProvider with its stub file content.

**Comment by stevemessick** [comment]
PTAL

Needed to do several things.
- Update test driver
- Ignore AndroidxTests
- Swap base and stub files for new class

See https://github.com/flutter/flutter-intellij/issues/4459 and #3583

**Pull request merged**